crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -519,11 +520,30 @@ impl<'w, 's> Commands<'w, 's> {
         self.queue.push(RemoveResource::<R>::new());
     }
 
+    /// Runs the supplied system on the [`World`] a single time.
+    ///
+    /// Calls [`SystemRegistry::run_system`](crate::system::RunSystem::run_system).
+    pub fn run_system<
+        M: Send + Sync + 'static,
+        S: IntoSystem<(), (), M> + Send + Sync + 'static,
+    >(
+        &mut self,
+        system: S,
+    ) {
+        self.queue.push(RunSystemCommand::new(system));
```

Comment:
My understanding is that this creates and initializes a new system every time you call `run_system` -- isn't this contrary to the idea of caching one-shot systems? Additionally, I don't like how there's less friction when using this method compared to `run_system_by_id`, which will encourage people to use the slower API.

Ideally, `run_system(my_system)` will only create and initialize a system if it's the first time it has been called for `my_system`, and subsequent call will use the cached system. I think you could achieve this by requiring `S: IntoSystemSet + IntoSystem`, and using the system set to check if the system has been initialized before.

Reply:
`run_system` was already added in #9718, where it was meant for testing purposes so the performance wasn't really considered then. I haven't used system sets before and I will look into it.

The reason why we did end up settling on this solution independently from #9718 was because relating a system `my_system` to a system in memory proved kinda unreliable (see: https://github.com/bevyengine/bevy/pull/8963#discussion_r1272987171), so instead of using `TypeId` we made `run_system` uncached and `run_by_id` uses a completely independent id system.

Reply:
> The reason why we did end up settling on this solution independently from #9718 was because relating a system `my_system` to a system in memory proved kinda unreliable (see: [#8963 (comment)](https://github.com/bevyengine/bevy/pull/8963#discussion_r1272987171)),

Fair enough, though I'm still not a fan of how the method names encourage people to use the sub-optimal function. Do we even need the version that runs the system and disposes of it? The original `run_system` function was relegated to a non-prelude trait for a reason -- we don't want to encourage anyone to be running systems like this.

Reply:
I agree that the less footgunny approach would be if we 'hid' `run_system` a bit more. In the original implementation, it's argued that it shouldn't be implemented for `App` as it dilutes the name space and it's equally simple to call `app.world.run_system(...)`. I think a simillair argument can be made for the `RunSystemCommand`. So if no-one has any objections, I think those two things can be removed.

Reply:
I meant #9366 this whole time, whoops.